docs/how-to/troubleshoot_deployment_issues.md

```diff
@@ -0,0 +1,168 @@
+| Blocked      |                                                  | Waiting for a relation | File a bug report     |
+
+Get the application logs:
+
+```shell
+$ microk8s.kubectl logs -f <pod_name> -n <model_name>
+```
+
+Attach the logs to the bug report.
+
+## 3. Terraform failed to deploy the Charmed Aether SD-Core because of selecting an existing Juju model
+
+### Symptoms
+
+Terraform may fail to deploy the Charmed Aether SD-Core modules because of the provided model name already exists in the Juju controller. When the `terraform apply -auto-approve` command is run, the deployment fails with the client error which express that the Juju model could not be created as following:
+
+```console
+Plan: 72 to add, 0 to change, 0 to destroy.
+juju_model.private5g: Creating...
+│ Error: Client Error
+│   with juju_model.private5g,
+│   on main.tf line 1, in resource "juju_model" "private5g":
+│    1: resource "juju_model" "private5g" {
+│ Unable to create model, got error: failed to open kubernetes client: annotations map[controller.juju.is/id:ced7016b-3a63-4133-8988-cf33068c3cdf
+│ model.juju.is/id:2abfe7ab-9e40-4e0d-8158-53450c47b2db] for namespace "private5g" not valid must include map[controller.juju.is/id:ced7016b-3a63-4133-8988-cf33068c3cdf
+│ model.juju.is/id:9755936b-8084-4397-8a67-28773b361dfa] (not valid)
+```
+
+### Recommended Actions
+
+Check all the existing Juju models:
+
+```shell
+$ juju models
+controller: microk8s-localhost
+
+Model       Cloud/Region        Type        Status      Units  Access  Last connection
+controller  microk8s/localhost  kubernetes  available   1       admin  just now
+private5g*  microk8s/localhost  kubernetes  destroying  -       admin  6 minutes ago
+private5g2  microk8s/localhost  kubernetes  destroying  -       admin  1 minute ago
+sdcore      microk8s/localhost  kubernetes  available   19      admin  2024-10-04
+```
+
+According to [Customize the model name using variables][Deploy SD-Core K8s with Terraform], provide a model name which does not exist in the Juju controller.
+
+## 4. Juju failed to deploy Charmed Aether SD-Core as Juju controller is not reachable
+
+### Symptoms
+
+While Juju is performing the deployment, the Juju controller may be unavailable which fails the deployment with the error `timeout to Juju Terraform provider` as following.
+
+```console
+$ terraform apply --auto-approve
+│ Error: Invalid provider configuration
+│ Provider "registry.terraform.io/juju/juju" requires explicit configuration. Add a provider block to the root module and configure the provider's required arguments as described in
+│ the provider documentation.
+│ Error: dial tcp 10.152.183.251:17070: i/o timeout
+│   with provider["registry.terraform.io/juju/juju"],
+│   on <empty> line 0:
+│   (source code not available)
+│ Connection error, please check the controller_addresses property set on the provider
+```
+
+### Recommended Actions
+
+Make sure that the Juju controller is available:
```

Comment:
```suggestion
Validate that the Juju controller is available:
```

Reply:
I do not see a big difference here, but replacing.